cf-macro/index.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+const _ = require('lodash')
+const Ajv = require('ajv')
+
+const alarms = require('slic-watch-core/alarms')
+const dashboard = require('slic-watch-core/dashboard')
+const CloudFormationTemplate = require('slic-watch-core/cf-template')
+const defaultConfig = require('slic-watch-core/default-config')
+const { slicWatchSchema } = require('slic-watch-core/config-schema')
+
+exports.handler = async function (event, context) {
+  let status = 'success'
+  let outputFragment = event.fragment
+  try {
+    const slicWatchConfig = (outputFragment.Metadata || {}).slicWatch || {}
+
+    if (slicWatchConfig.enabled !== false) {
+      const ajv = new Ajv({
+        unicodeRegExp: false
+      })
+      const config = _.merge(defaultConfig, slicWatchConfig)
+
+      const slicWatchValidate = ajv.compile(slicWatchSchema)
+      const slicWatchValid = slicWatchValidate(slicWatchConfig)
+
+      if (!slicWatchValid) {
+        throw new Error('SLIC Watch configuration is invalid: ' + ajv.errorsText(slicWatchValidate.errors))
+      }
+
+      const alarmActions = []
+
+      if (slicWatchConfig.topicArn) {
+        alarmActions.push(slicWatchConfig.topicArn)
+      } else if (process.env.SNSTopic) {
+        alarmActions.push(process.env.SNSTopic)
+      }
+
+      const context = {
+        region: event.region,
+        stackName: (event.templateParameterValues.stack) ? event.templateParameterValues.stack : '',
+        alarmActions
+      }
+
+      const cfTemplate = CloudFormationTemplate(
+        outputFragment
+      )
+      const serverless = { cli: console }
+      const functionAlarmConfigs = {}
+      const functionDashboardConfigs = {}
+
+      const lambdaResources = cfTemplate.getResourcesByType(
+        'AWS::Lambda::Function'
+      )
+
+      for (const [funcResourceName, funcResource] of Object.entries(lambdaResources)) {
+        const functionName = funcResource.Properties.FunctionName ? funcResource.Properties.FunctionName : { Ref: funcResourceName }
+        const funcConfig = funcResource.Metadata.slicWatch || {}
+        functionAlarmConfigs[functionName] = funcConfig.alarms || {}
+        functionDashboardConfigs[functionName] = funcConfig.dashboard || {}
+      }
+
+      const alarmService = alarms(serverless, config.alarms, functionAlarmConfigs, context)
+      alarmService.addAlarms(cfTemplate)
+      const dashboardService = dashboard(serverless, config.dashboard, functionDashboardConfigs, context)
```

Comment:
It seems a little bit weird that here we are passing a mock `serverless` object. Should we maybe re-consider in a future version of this to refactor the `alarms` and `dashboard` function to not depend on a `serverless` instance (assuming that's possible) so we can clean up here?

If you agree with this, I would suggest only to add a TODO here and maybe open an issue

Reply:
@ssamueluk I think it makes sense to remove the mock `serverless` object and replace with a generic logger